src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotUseInsecureDeserializerMethodsBase.cs

```diff
@@ -70,12 +88,17 @@ public sealed override void Initialize(AnalysisContext context)
                             if (invocationOperation.Instance?.Type?.DerivesFrom(deserializerTypeSymbol) == true
                                 && cachedDeserializationMethodNames.Contains(invocationOperation.TargetMethod.MetadataName))
                             {
-                                operationAnalysisContext.ReportDiagnostic(
-                                    Diagnostic.Create(
-                                        this.MethodUsedDescriptor,
-                                        invocationOperation.Syntax.GetLocation(),
-                                        invocationOperation.TargetMethod.ToDisplayString(
-                                            SymbolDisplayFormat.MinimallyQualifiedFormat)));
+                                DiagnosticDescriptor? chosenDiagnostic =
+                                    this.ChooseDiagnosticDescriptor(operationAnalysisContext, wellKnownTypeProvider);
+                                if (chosenDiagnostic != null)
+                                {
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        Diagnostic.Create(
+                                            chosenDiagnostic,
+                                            invocationOperation.Syntax.GetLocation(),
+                                            invocationOperation.TargetMethod.ToDisplayString(
```

Comment:
I don't know what's the rule but I remember to have had feedback about not using `ToDisplayString` in diagnostic.

Reply:
I believe this pattern is followed across all security rules, so @dotpaul is being consistent here.